### PR TITLE
chore(v3): Update PR template

### PR DESCRIPTION
**Pull request by traeok**
## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by anaxceron — APPROVED** [review]
thanks @traeok!

**Review by JillieBeanSim — COMMENTED** [review]
Thanks @traeok especially for updating all the pnpm refs to be the same 😄 I did leave a couple comments

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -14,29 +14,37 @@ Changelog:
...
+
+**General**
 
 - [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
-- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
-- [ ] PR Description is included
-- [ ] gif or screenshot is included if visual changes are made
+- [ ] All PR dependencies have been merged and published (if applicable)
+- [ ] A GIF or screenshot is included in the PR for visual changes
 - [ ] `pnpm --filter vscode-extension-for-zowe vscode:prepublish` has been executed
```
Comment (JillieBeanSim) [inline .github/PULL_REQUEST_TEMPLATE.md:35]: I think the one shown on PRs comes from the main branch, should we include both with Zowe version in the main branch PR?

**Comment by traeok** [comment]
Closing in favor of #2954 as the changes will apply to the `next` branch automatically

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -14,29 +14,37 @@ Changelog:
...
+_Put an `x` in all boxes that apply._
 
-- [ ] Bugfix (non-breaking change which fixes an issue)
-- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] Enhancement (non-breaking change which adds or improves functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-- [ ] Updates to Documentation or Tests (if none of the other choices apply)
+- [ ] Documentation (Markdown, README updates)
+- [ ] Test cases (unit, integration, and/or end-to-end tests)
```
[the remark this answers is not in the document]
Comment (traeok) [reply .github/PULL_REQUEST_TEMPLATE.md:23]: Resolved in v2 PR 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -14,29 +14,37 @@ Changelog:
...
-- [ ] I have added unit test and it is passing
-- [ ] I have added integration test and it is passing
+
+**Code coverage**
+
 - [ ] There is coverage for the code that I have added
-- [ ] I have tested it manually and there are no regressions found
-- [ ] I have added necessary documentation (if appropriate)
-- [ ] Any PR dependencies have been merged and published (if appropriate)
+- [ ] I have added new test cases and they are passing
```
[the remark this answers is not in the document]
Comment (traeok) [reply .github/PULL_REQUEST_TEMPLATE.md:42]: If you are referring to line 40, I intended the item "There is coverage for the code added" to refer to existing coverage (as that's how it comes across), and this line refers to new coverage that was added. We can shrink both of these down to one item "There is existing or new coverage for the added code" to improve readability if you prefer that 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -14,29 +14,37 @@ Changelog:
...
+- [ ] I have manually tested the changes
+
+**Deployment**
+
+- [ ] I have added developer documentation (if applicable)
+- [ ] Documentation should be added to Zowe Docs
```
[the remark this answers is not in the document]
Comment (traeok) [reply .github/PULL_REQUEST_TEMPLATE.md:48]: Going to tackle this once I get Ana's feedback as I don't want to add her to the PR template without permission; there might be a preferred method to reach out to the Doc Squad for adding ZE documentation, will follow up on that in the v2 PR.

**Pull request closed without merging**